penguin/engine.py

```
@@ -2520,36 +2555,46 @@ async def _execute_codeact_actions(
             return action_results
 
         actions: List[CodeActAction] = parse_action(assistant_response)
+        tool_calls = tool_calls_from_codeact_actions(actions)
         logger.debug(
-            "[AUTO-CONTINUE FIX] Parsed %s actions from response", len(actions)
+            "[AUTO-CONTINUE FIX] Parsed %s actions from response", len(tool_calls)
+        )
+
+        async def _execute_actionxml_call(tool_call):
```

Ruff (0.15.11)

[warning] Missing return type annotation for private function `_execute_actionxml_call`

(ANN202)